pyxrf/model/fileio.py

```diff
@@ -749,34 +809,90 @@ def render_data_to_gui(runid):
+
     data_sets = OrderedDict()
     img_dict = OrderedDict()
-    fname = 'scan2D_{}'.format(runid)
 
-    data_out = fetch_data_from_db(runid)
-
-    # Transfer to standard format pyxrf GUI can take
-    fname_sum = fname+'_sum'
+    data_from_db = fetch_data_from_db(runid,
+                                      # Always create unique file name by adding
+                                      #   version number
+                                      fname_add_version=True,
+                                      # Always load data from all detectors
+                                      create_each_det=True,
+                                      # Always create data file (processing results
+                                      #   are going to be saved in the file)
+                                      output_to_file=True)
+
+    if not len(data_from_db):
+        logger.warning(f"No detector data was found in Scan #{runid}")
+        return
+    else:
+        logger.info(f"Data from {len(data_from_db)} detectors were found in Scan #{runid}.")
+        if len(data_from_db) > 1:
+            logger.warning(f"Selecting only the first dataset from Scan #{runid}.")
+
+    # If the experiment contains data from multiple detectors (for example two separate
+    #   Xpress3 detectors) that need to be treated separately, only the data from the
+    #   first detector is loaded. Data from the second detector is saved to file and
+    #   can be loaded from the file. Currently this is a very rare case (only one set
+    #   of such experiments from SRX beamline exists).
+    data_out = data_from_db[0]['dataset']
+    fname = data_from_db[0]['file_name']
+    detector_name = data_from_db[0]['detector_name']
+
+    # Create file name for the 'sum' dataset ('file names' are used as dictionary
+    #   keys in data storage containers, as channel labels in plot legends,
+    #   and as channel names in data selection widgets.
+    #   Since there is currently no consistent metadata in the start documents
+    #   and/or data files, let's leave original labeling conventions for now.
+    fname_no_ext = os.path.splitext(os.path.basename(fname))[0]
+    fname_sum = fname_no_ext + '_sum'
+
+    # Determine the number of available detector channels and create the list
+    #   of channel names. The channels are named as 'det1', 'det2', 'det3' etc.
+    xrf_det_list = [nm for nm in data_out.keys() if 'det' in nm and 'sum' not in nm]
+
+    det_sum = None
     if 'det_sum' in data_out:
-        det_sum = data_out['det_sum']
+        det_sum = np.copy(data_out['det_sum'][:, :, 0:spectrum_cut])
     else:
-        det_sum = data_out['det1'] + data_out['det2'] + data_out['det3']
+        for det_name in xrf_det_list:
+            if det_sum is None:
+                det_sum = np.array(data_out[det_name][:, :, 0:spectrum_cut])
+            else:
+                det_sum += data_out[det_name][:, :, 0:spectrum_cut]
```

Comment:
``data_out`` is already the numpy array. We do need a copy in line 862. As for line 858, I think that it is safer to make a copy, since ``data_sum`` is treated separately later in the program. I think that making a copy here may help avoid potential problems in the future.